features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/components/APIDetailsTopMenu.jsx

```diff
@@ -43,55 +44,62 @@ const styles = theme => ({
-                <div className={classes.backText}>
-                    BACK TO <br />
-                    LISTING
-                </div>
-            </Link>
-            <VerticalDivider height={70} />
-            <ThumbnailView api={api} width={70} height={50} />
-            <div style={{ marginLeft: theme.spacing.unit }}>
-                <Typography variant='display1'>
-                    {api.name} : {api.version}
-                </Typography>
-                <Typography variant='caption' gutterBottom align='left'>
-                    Created by: {api.provider}
-                </Typography>
-            </div>
-            <VerticalDivider height={70} />
-            <div className={classes.infoItem}>
-                <Typography variant='subheading' gutterBottom>
-                    {api.lifeCycleStatus}
-                </Typography>
-                <Typography variant='caption' gutterBottom align='left'>
-                    State
-                </Typography>
-            </div>
+        <ApiContext.Consumer>
+            {({ api }) => (
+                <div className={classes.root}>
+                    <Link to='/apis' className={classes.backLink}>
+                        <KeyboardArrowLeft className={classes.backIcon} />
+                        <div className={classes.backText}>
+                            BACK TO <br />
+                            LISTING
+                        </div>
+                    </Link>
+                    <VerticalDivider height={70} />
+                    <ThumbnailView api={api} width={70} height={50} />
+                    <div style={{ marginLeft: theme.spacing.unit }}>
+                        <Typography variant='display1'>
+                            {api.name} : {api.version}
+                        </Typography>
+                        <Typography variant='caption' gutterBottom align='left'>
+                            Created by: {api.provider}
+                        </Typography>
+                    </div>
+                    <VerticalDivider height={70} />
+                    <div className={classes.infoItem}>
+                        <Typography variant='subheading' gutterBottom>
+                            {api.lifeCycleStatus}
+                        </Typography>
+                        <Typography variant='caption' gutterBottom align='left'>
+                            State
+                        </Typography>
+                    </div>
 
-            <VerticalDivider height={70} />
+                    <VerticalDivider height={70} />
 
-            <a target='_blank' rel='noopener noreferrer' href={storeURL} className={classes.viewInStoreLauncher}>
-                <div>
-                    <LaunchIcon />
+                    <a
+                        target='_blank'
+                        rel='noopener noreferrer'
+                        href={`${window.location.origin}/store-new/apis/${api.id}/overview`}
```

Comment:
appcontext (store-new) need to be read from a config file. If not will break in reverse proxy scenarios.

Reply:
We need to do this across the store-new and publisher-new apps and fix them there are many places this is hard coded.